packages/vaadin-themable-mixin/src/css-utils.js

```diff
@@ -15,16 +15,14 @@ import { adoptStyles } from 'lit';
  * @return {CSSStyleSheet[]}
  */
 function getEffectiveStyles(component) {
-  const componentClass = component.constructor;
+  const { baseStyles, themeStyles, elementStyles, lumoInjector } = component.constructor;
+  const lumoStyleSheet = component.__lumoStyleSheet;
 
-  const styleSheet = component.__lumoInjectorStyleSheet;
-  if (styleSheet) {
-    return (componentClass.baseStyles ?? componentClass.themeStyles)
-      ? [...componentClass.baseStyles, styleSheet, ...componentClass.themeStyles]
-      : [styleSheet, ...componentClass.elementStyles];
+  if (lumoStyleSheet && (baseStyles || themeStyles)) {
+    return [...(lumoInjector.includeBaseStyles ? baseStyles : []), lumoStyleSheet, ...themeStyles];
   }
 
-  return componentClass.elementStyles;
+  return [lumoStyleSheet, ...elementStyles].filter(Boolean);
```

Comment:
Seems all tests pass if I remove `lumoStyleSheet` from here and just change this line to:

```suggestion
  return elementStyles;
```

Not sure if there can be a case where we have Lumo stylesheet but no base styles.

Reply:
UPD: the logic is actually needed for cases where a component doesn't extend `ThemableMixin` which is the case for `vaadin-combo-box-scroller`. This could be covered by a dedicated test for such a component.